source/client/renderer/TileRenderer.cpp

```diff
@@ -120,12 +120,16 @@ float TileRenderer::getWaterHeight(const TilePos& pos, const Material* pCheckMtl
 
 bool TileRenderer::canRender(int renderShape)
 {
-	return renderShape == SHAPE_SOLID || renderShape == SHAPE_STAIRS;
+	return renderShape == SHAPE_SOLID || renderShape == SHAPE_STAIRS || renderShape == SHAPE_FENCE || renderShape == SHAPE_CACTUS;
 }
 
 // @NOTE: This sucks! Very badly! But it's how they did it.
 void TileRenderer::renderEast(Tile* tile, const Vec3& pos, int texture)
 {
+	if (tile->getRenderShape() == SHAPE_CACTUS)
+	{
+		tile->setShape(0.0625, 0, 0, 0.9375, 1, 1);
+	}
```

Comment:
Consider not using a full set of curly braces around one statement. Also, put a new line between the if-statement and the static constexpr float.